test/e2e/multiple_zk_test.go

```diff
@@ -0,0 +1,187 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package e2e
+
+import (
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	api "github.com/pravega/zookeeper-operator/pkg/apis/zookeeper/v1beta1"
+	zk_e2eutil "github.com/pravega/zookeeper-operator/pkg/test/e2e/e2eutil"
+)
+
+// Test create and recreate a Pravega cluster with the same name
+func testMultiZKCluster(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	doCleanup := true
+	ctx := framework.NewTestCtx(t)
+	defer func() {
+		if doCleanup {
+			ctx.Cleanup()
+		}
+	}()
+
+	namespace, err := ctx.GetNamespace()
+	g.Expect(err).NotTo(HaveOccurred())
+	f := framework.Global
+
+	defaultCluster := zk_e2eutil.NewDefaultCluster(namespace)
+
+	defaultCluster.WithDefaults()
+	defaultCluster.Status.Init()
+	defaultCluster.ObjectMeta.Name = "zk1"
+	defaultCluster.Spec.Persistence.VolumeReclaimPolicy = "Delete"
+
+	zk1, err := zk_e2eutil.CreateCluster(t, f, ctx, defaultCluster)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	// A default Pravega cluster should have 2 pods: 1 controller, 1 segment store
+	podSize := 3
+	err = zk_e2eutil.WaitForClusterToBecomeReady(t, f, ctx, zk1, podSize)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	defaultCluster = zk_e2eutil.NewDefaultCluster(namespace)
+
+	defaultCluster.WithDefaults()
+	defaultCluster.Status.Init()
+	defaultCluster.Spec.Persistence.VolumeReclaimPolicy = "Delete"
+	defaultCluster.ObjectMeta.Name = "zk2"
+	initialVersion := "0.2.5"
+	upgradeVersion := "0.2.7"
+	defaultCluster.Spec.Image = api.ContainerImage{
+		Repository: "pravega/zookeeper",
+		Tag:        initialVersion,
+	}
+	zk2, err := zk_e2eutil.CreateCluster(t, f, ctx, defaultCluster)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	err = zk_e2eutil.WaitForClusterToBecomeReady(t, f, ctx, zk2, podSize)
+	g.Expect(err).NotTo(HaveOccurred())
+	// This is to get the latest Pravega cluster object
+	zk2, err = zk_e2eutil.GetCluster(t, f, ctx, zk2)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	g.Expect(zk2.Status.CurrentVersion).To(Equal(initialVersion))
+
+	defaultCluster = zk_e2eutil.NewDefaultCluster(namespace)
+
+	defaultCluster.WithDefaults()
+	defaultCluster.Status.Init()
+	defaultCluster.ObjectMeta.Name = "zk3"
+	defaultCluster.Spec.Persistence.VolumeReclaimPolicy = "Delete"
+
+	zk3, err := zk_e2eutil.CreateCluster(t, f, ctx, defaultCluster)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	err = zk_e2eutil.WaitForClusterToBecomeReady(t, f, ctx, zk3, podSize)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	// This is to get the latest zk cluster object
+	zk1, err = zk_e2eutil.GetCluster(t, f, ctx, zk1)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	// scale up the replicas in first cluster
+	zk1.Spec.Replicas = 5
+	podSize = 5
+
+	err = zk_e2eutil.UpdateCluster(t, f, ctx, zk1)
+	g.Expect(err).NotTo(HaveOccurred())
```

Comment:
An assertion can be added here to check whether`Spec.Replicas` = 5 or not

Reply:
that is being checked in the next call `waitforclustertobecomeready`